### PR TITLE
Port away from Q_FOREACH macro

### PR DESCRIPTION
**Pull request by alexfikl**
This switches to C++11 ranged for-loops instead. The `Q_FOREACH` macro has been deprecated for a while.

### CODE REVIEW

File: app/configdialog.cpp [hunk 1]
```diff
@@ -82,8 +82,9 @@ void ConfigDialog::writeSettings()
 
 void ConfigDialog::setTranslatedHighlightTypeNames(const QStringList &typeNames)
 {
-    Q_FOREACH (const QString &typeName, typeNames)
```
Comment (jfmcarreira) [inline app/configdialog.cpp:85]: I know it is not related, but what you think of using auto.

Instead of types use: `const auto&`

Comment (alexfikl) [reply app/configdialog.cpp:85]: Yeah, sure, that sounds good to me! I'll change them in these for loops.

Comment (alexfikl) [reply app/configdialog.cpp:85]: Done in [8086ab3](https://github.com/fhackenberger/ktikz/pull/64/commits/8086ab3e35ae03e298a25d1f62c9c51093029698).

**Comment by jfmcarreira** [comment]
I squashed your three commits and merged

**Pull request merged**